src/dotnet-core-uninstall/dotnet-core-uninstall.csproj

```diff
@@ -37,4 +37,20 @@
     </EmbeddedResource>
   </ItemGroup>
 
+  <Target Name="PublishMacOSUniversalBinary">
+    <ItemGroup>
+      <_Rids Include="osx-x64;osx-arm64" />
+      <_RidSpecificBuild Include="$(MSBuildThisFile)" AdditionalProperties="RuntimeIdentifier=%(_Rids.Identity);PublishSingleFile=true;Configuration=Release" />
+    </ItemGroup>
+
+    <MSBuild Projects="@(_RidSpecificBuild)" BuildInParallel="true" Targets="Publish;PublishItemsOutputGroup">
+     <Output TaskParameter="TargetOutputs" ItemName="_RidSpecificSingleFile" />
+    </MSBuild>
+    <PropertyGroup>
+      <_LipoOuputPath>$(PublishDir)\osx-universal\$(AssemblyName)</_LipoOuputPath>
+    </PropertyGroup>
+
+    <Exec Command="lipo @(_RidSpecificSingleFile, ' ') -create -output $(_LipoOuputPath)" Condition="$([MSBuild]::IsOSPlatform('OSX'))" />
```

Comment:
I don't know if you care, but this won't work if any of these paths have spaces in them.